File: 2019-01-18/9375/9375.py
import sys

info = {}
T = int(sys.stdin.readline())

result = 1

for i in range(0, T):
	n = int(sys.stdin.readline())
	for j in range(0, n):
		order = sys.stdin.readline().split()
		if order[1] not in info:
			info[order[1]] = 1
		else: 
			info[order[1]] += 1
	for k in info:	
		result*=(info[k]+1)

	print(result-1)
	result=1
	info.clear()		
		
	

File: 2019-01-18/9375/9375.cpp
#include <cstdio>
#include <string>
#include <map>
#include <iostream>
using namespace std;

map <string, int>info;
string order[102];
string type[102];
int main(){
	int T, n;
	scanf("%d", &T);
	int result = 1;
	for(int i =0; i < T; i++){
		scanf("%d", &n);
		for(int i =0; i < n; i++){
			cin >> order[i] >> type[i];
			if(info.find(type[i]) == info.end())
				info[type[i]] = 1;
			else
				info[type[i]] += 1;
		}
		for (map<string,int>::iterator j= info.begin(); j != info.end(); j++)
			 result *= j->second+1;
			 
		cout << result-1<<endl;	
		info.clear();
		result = 1;
	}
}